src/Razor/src/Microsoft.NET.Sdk.Razor/ApplicationPartsProvider.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class ApplicationPartsProvider
+    {
+        private readonly HashSet<string> _mvcAssemblies;
+        private readonly Dictionary<string, AssemblyItem> _lookup = new Dictionary<string, AssemblyItem>(StringComparer.Ordinal);
+
+        public ApplicationPartsProvider(IReadOnlyList<string> mvcAssemblies, IReadOnlyList<ResolveReferenceItem> resolveReferenceResult)
```

Comment:
Is there any reason why this wouldn't just be a static method that returns the results?

Reply:
I override the part that goes to disk as part of unit tests. I could have this take a `Func`, but this is way cleaner to work with.